core/src/test/scala/unit/kafka/server/ReplicaManagerTest.scala

```diff
@@ -3624,14 +3624,18 @@ class ReplicaManagerTest {
     val remoteLogManagerConfig = new RemoteLogManagerConfig(config)
     val mockLog = mock(classOf[UnifiedLog])
     val brokerTopicStats = new BrokerTopicStats(java.util.Optional.of(KafkaConfig.fromProps(props)))
+    val mockMetadataCache = mock(classOf[MetadataCache])
     val remoteLogManager = new RemoteLogManager(
       remoteLogManagerConfig,
       0,
       TestUtils.tempRelativeDir("data").getAbsolutePath,
       "clusterId",
       time,
       _ => Optional.of(mockLog),
-      brokerTopicStats)
+      brokerTopicStats,
+      mockMetadataCache
+    )
+    when(remoteLogManager.metadataCache().getTopicId(topic)).thenReturn(topicId)
```

Comment:
What's the difference between `mockMetadataCache` and `remoteLogManager.metadataCache()`? If no difference, could we use the former one and remove the `remoteLogManager#metadataCache()` method?